Article/Article.js

```diff
@@ -1,20 +1,63 @@
-// Because classes are not hoised you will need to start your code at the bottom of the page.  Look for the comment "START HERE"
+// Because classes are not hoisted you will need to start your code at the bottom of the page.  Look for the comment "START HERE"
 
 class Article {
   constructor(domElement) {
     // assign this.domElement to the passed in domElement
-    this.domElement;
-    // create a reference to the ".expandButton" class. 
-    this.expandButton;
+    this.domElement = domElement;
+    // create a reference to the ".expandButton" class.
+    this.expandButton = domElement.querySelector(".expandButton");
+    // read button
+    this.readButton = domElement.querySelector(".readButton");
+    // console.log(this.readButton);
     // Using your expandButton reference, update the text on your expandButton to say "expand"
-    
+    this.expandButton.textContent = "Click to Expand";
+
     // Set a click handler on the expandButton reference, calling the expandArticle method.
-    
+    this.expandButton.addEventListener("click", () => this.expandArticle());
+    // event listener calling removeArticle
+    this.readButton.addEventListener("click", () => this.removeArticle());
+  }
+
+  removeArticle() {
+    this.domElement.classList.add("read");
   }
 
   expandArticle() {
     // Using our reference to the domElement, toggle a class to expand or hide the article.
-
+    // articles.forEach(article => {
+    //   if (article.classList.contains("article-open")) {
+    //     article.classList.remove("article-open");
+    //   }
+    // });
+    // this.domElement.classList.toggle("article-open");
+    const arrArticles = Array.from(articles);
+    arrArticles.splice(arrArticles.indexOf(this.domElement), 1);
+    // console.log(arrArticles);
+    // looping through all the articles and closing each of them
+    for (let i = 0; i < arrArticles.length; i++) {
+      arrArticles[i].classList.remove("article-open");
+      arrArticles[i].classList.add("article-close");
+    }
+    // all the if else is basically toggling the text to close and the class
```

Comment:
I appreciate the comments like these that help explain your mindset with the code.